scc/src/opts.rs

```diff
@@ -0,0 +1,147 @@
+                *fixed_args.last_mut().unwrap() = format!("{} {}", last, &parts.pop_front().unwrap());
+                for part in parts {
+                    fixed_args.push(part.to_string());
+                }
+            } else {
+                let last = fixed_args.last().unwrap();
+                *fixed_args.last_mut().unwrap() = format!("{} {}", last, broken_arg);
+            }
+        }
+        if contains_quote {
+            broken = !broken;
+        }
+    }
+
+    #[cfg(test)]
+    info!("Fixed args: {:#?}", fixed_args);
+
+    fixed_args
+}
+
+#[derive(Clone, Debug)]
+pub struct Opts {
+    pub script_paths: Vec<PathBuf>,
+    pub cache_file: PathBuf,
+    pub cache_dir: Option<PathBuf>,
+    pub optimize: bool,
+    pub threads: Option<u8>,
+    pub no_warnings: bool,
+    pub no_testonly: bool,
+    pub no_breakpoint: bool,
+    pub profile_off: bool
+}
+
+fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<bool> {
+    any::<String>(name)
+        .help(help)
+        .guard(|s| s.starts_with('-'), "doesn't start with -")
+        .parse(move |s| {
+            let (state, cur_name) = if let Some(rest) = s.strip_prefix('-') {
+                (true, rest)
+            } else {
+                return Err(format!("{} is not a toggle option", s));
+            };
+            if cur_name != name {
+                Err(format!("{} is not a known toggle option name", cur_name))
+            } else {
+                Ok(state)
+            }
+        })
+        .anywhere()
+}
+
+fn slong(tag_str: &'static str, arg_str: &'static str) -> impl Parser<String> {
+    let tag = any::<String>(tag_str)
+        .guard(move |s| s.as_str() == tag_str, "not arg name");
+    let value = positional::<String>(arg_str);
+    construct!(tag, value)
+        .anywhere()
+        .map(|pair| pair.1)
+}
+
+fn script_paths() -> impl Parser<Vec<PathBuf>> {
+    let tag = any::<String>("-compile")
+        .guard(|s| s == "-compile", "not compile");
+    let value = positional::<OsString>("SCRIPT_PATH");
+    construct!(tag, value)
+        .anywhere()
+        .map(|pair| pair.1)
+        .map(PathBuf::from)
+        .many()
+        .catch()
+}
+
+fn cache_dir() -> impl Parser<Option<PathBuf>> {
+    let tag = any::<String>("-customCacheDir")
+        .guard(|s| s == "-customCacheDir", "not customCacheDir");
+    let value = positional::<OsString>("CACHE_DIR");
+    construct!(tag, value)
+        .anywhere()
+        .map(|pair| pair.1)
+        .map(PathBuf::from)
+        .optional()
+        .catch()
+}
+
+impl Opts {
+    pub fn load(args: Vec<String>) -> Self {
+        let cache_file = any::<OsString>("CACHE_FILE")
+            .anywhere()
+            .map(PathBuf::from);
+        let optimize = toggle_options("optimize", "Optimize the redscripts");
+        let threads = slong("-threads", "Number of theads to script_paths on").map(|s| s.parse::<u8>().unwrap()).optional();
```

Comment:
I think you need to use `.parse` for it to propagate errors correctly
```suggestion
        let threads = slong("-threads", "Number of theads to script_paths on").parse(|s| s.parse::<u8>()).optional();
```